cpp/src/io/cudftable.cpp

```diff
@@ -38,6 +38,15 @@ struct cudftable_header {
   uint32_t format_version;   ///< Format version number
   uint64_t metadata_length;  ///< Length of metadata buffer in bytes
   uint64_t data_length;      ///< Length of data buffer in bytes
+
+  cudftable_header() = default;
```

Comment:
The default constructor here probably does not do anything, as I just did some search:
> default-initialized (for fundamental types like int, that means they are left uninitialized, with indeterminate values).


So imagine that if we do this:
```
{
  auto header = cudftable_header(meta_size, data_size);
  // destroy `header`
}
{
  auto new_header = cudftable_header{};
...
```
In the last line, it is likely that the content of `new_header` will be just exactly the same as the destroyed `header` object. That can potentially cause problem later on.

So probably an explicit constructor to set zero for everything is better:
```
cudftable_header(): magic{0}, ..... {}
```